msticpy/data/queries/kql_sent_linuxsyslog_logon.yaml

```diff
@@ -0,0 +1,81 @@
+metadata:
+  version: 1
+  description: Linux Syslog Logon Data Queries
+  data_environments: [LogAnalytics]
+  data_families: [LinuxSyslog]
+  tags: ['linux', 'syslog']
+defaults:
+  metadata:
+    data_source: 'linux_syslog'
+  parameters:
+      table:
+        description: Table name
+        type: str
+        default: 'Syslog'
+      start:
+        description: Query start time
+        type: datetime
+      end:
+        description: Query end time
+        type: datetime
+      user:
+        description: Optional user to filter on
+        type: str
+        default: ''
+      add_query_items:
+        description: Additional query clauses
+        type: str
+        default: ''
+      subscription_filter:
+        description: Optional subscription/tenant filter expression
+        type: str
+        default: 'true'
+      query_project:
+        description: Column project statement
+        type: str
+        default: '
+          | project
+            TenantId,
+            Computer,
+            Facility,
+            TimeGenerated,
+            HostName,
+            SeverityLevel,
+            SyslogMessage,
+            ProcessID,
+            ProcessName,
+            HostIP,
+          | extend TimeCreatedUtc=TimeGenerated'
+sources:
+  user_logon:
+    description: All user logon events
+    args:
+      query: '
+        {table}
+        | where {subscription_filter}
+        | where TimeGenerated >= datetime({start})
+        | where TimeGenerated <= datetime({end})
+        | where Computer == "{host_name}"
```

Comment:
Usually I've been taking host_name to mean unqualified host, not fqdn. I think in some cases I've used the "has" operator (Computer has host_name) since that will find a hostname in a delimited string. Host entities also have a Computer attribute that is the fqdn (or whatever the full computer name is in the source event). In many cases this won't matter (other than being confusing to have different definitions) but when we call a query like this qry.get_data(some_alert_or_event), the code will specifically look for the host_name attribute.
We should probably extend our entities to add host_name, account_name, etc. properties to support queries passing these entity types as args.

Reply:
Good point on the fqdn/host issue. I have swapped these all to 'has'. I will take a look at the entity extension at some point as that would certainly make things a bit simpler.